src/main/java/com/yanna/stepanova/controller/BookController.java

```diff
@@ -17,44 +20,59 @@
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book manager", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/books")
 public class BookController {
     private final BookService bookService;
 
     @PostMapping
+    @Operation(summary = "Create a new book",
+            description = "Create a new book entity in the database")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get a book by id",
+            description = "Get a book entity by id from the database")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @GetMapping("/by-author")
+    @Operation(summary = "Get all books by an author",
+            description = "Get all books filtered by an author")
     public List<BookDto> getAllByAuthor(@RequestParam String author) {
         return bookService.getAllByAuthor(author);
     }
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    @Operation(summary = "Get all books in parts",
+            description = "Get all the books in parts + using sorting")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
 
     @GetMapping("/search")
+    @Operation(summary = "Search all books by parameters",
+            description = "Search all books by parameters: titles and authors")
     public List<BookDto> searchBooks(BookSearchParams searchParams) {
         return bookService.search(searchParams);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update a book by id",
+            description = "Update a book with new data by id in the database")
     public BookDto updateBookById(@PathVariable Long id,
                                   @RequestBody CreateBookRequestDto newRequestDto) {
         return bookService.updateBook(id, newRequestDto);
     }
 
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete a book by id",
+            description = "Delete a book by id (not physically - just mark it as deleted)")
     public String delete(@PathVariable Long id) {
```

Comment:
```suggestion
    public String delete(@PathVariable @Positive Long id) {
```
Don't forget to add `@Validated` under the class